### PR TITLE
tetgen: add recipe

### PR DESCRIPTION
**Pull request by nehaljwani**

### CODE REVIEW

File: recipes/tetgen/bld.bat [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+:: Configure using the CMakeFiles
+%LIBRARY_BIN%\cmake -G "NMake Makefiles" ^
+    -D CMAKE_BUILD_TYPE=Release ^
+    -D CMAKE_PREFIX_PATH=%LIBRARY_PREFIX% ^
+    -D CMAKE_INSTALL_PREFIX:PATH=%LIBRARY_PREFIX% ^
+    %SRC_DIR%
+if errorlevel 1 exit 1
+
+:: Build!
+nmake
```
Comment (ocefpaf) [inline recipes/tetgen/bld.bat]: Is there a `ctest` or something similar we can run here?

File: recipes/tetgen/build.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+make tetgen tetlib
```
Comment (ocefpaf) [inline recipes/tetgen/build.sh]: Is there a `make check` or something similar we can run here?

File: recipes/tetgen/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: http://wias-berlin.de/software/tetgen/1.5/src/tetgen{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  features:
+    - vc9  # [win and py27]
+    - vc14  # [win and (py35 or py36)]
```
Comment (ocefpaf) [inline recipes/tetgen/meta.yaml]: Even though we are not building `vc10` (Python 3.4) in conda-forge it is nice to leave the `vc 10  # [win and py34]` here so others can use the recipe.

File: recipes/tetgen/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+requirements:
+  build:
+    - python  # [win]
+    - toolchain
+    - cmake >2.8  # [win]
+
```
Comment (ocefpaf) [inline recipes/tetgen/meta.yaml]: Add a run section with `vc 9  # [win and py27]`, etc.

File: recipes/tetgen/run_test.py [hunk 5]
```diff
@@ -0,0 +1,2 @@
+# this file exists only to trick the conda-build test env creator to install
```
Comment (ocefpaf) [inline recipes/tetgen/run_test.py]: If you add the run section you can remove this file.

File: recipes/tetgen/bld.bat (hunk 1 above)
Comment (nehaljwani) [inline recipes/tetgen/bld.bat]: Unfortunately, no :sob:

File: recipes/tetgen/build.sh (hunk 2 above)
Comment (nehaljwani) [inline recipes/tetgen/build.sh]: Unfortunately, no :cry:

File: recipes/tetgen/meta.yaml (hunk 4 above)
Comment (nehaljwani) [inline recipes/tetgen/meta.yaml]: Updated!

**Pull request merged**